src/general/DialogHelper.cs

```diff
@@ -0,0 +1,22 @@
+using Godot;
+
+/// <summary>
+///   This is the static helper class used to handle default dialog processes.
+/// </summary>
+public static class DialogHelper
+{
+    /// <summary>
+    ///   This is the static helper for dialog escape input.
+    /// </summary>
+    /// <returns>True if handled, false if not.</returns>
+    public static bool DefaultDialogEscapeHandler(WindowDialog dialog)
+    {
+        if (!dialog.Visible)
+        {
+            return false;
+        }
+
+        dialog.Hide();
+        return true;
```

Comment:
The code here can just be directly put inside the custom popup class. Also since normal popups already closes by themselves if you press any key (if I recall correctly) I'd amend the check on top by adding two more conditions that only exclusive _and_ escapable-allowed dialogs can be closed with escape, meaning there would be a bool like `AllowExclusiveCloseWithEscape` in `CustomDialog`.